diff --git a/logappend.c b/logappend.c
index def00c1..20bfff8 100755
--- a/logappend.c
+++ b/logappend.c
@@ -169,6 +169,9 @@ CmdLineResult parse_cmdline(int argc, char *argv[], int is_batch) {
       case 'R':
         //room ID
         R.roomID = atoi(optarg);
+        if (R.roomID < 0 || R.roomID > 1073741823){
+            invalidinput = true;
+        }
         nonbatch= true;
         break;
 
